pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -681,6 +681,13 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName, asName s
 			vmSpec.ManagedIdentity = azure.GenerateManagedIdentityName(s.scope.SubscriptionID, s.scope.MachineConfig.ResourceGroup, s.scope.MachineConfig.ManagedIdentity)
 		}
 
+		if s.scope.MachineConfig.CapacityReservationGroupID != "" {
+			if err = validateCapacityReservationGroupID(s.scope.MachineConfig.CapacityReservationGroupID); err != nil {
+				return fmt.Errorf("invalid capacityReservationGroupID : %w", err)
```

Comment:
the error message should be describe as a failure event. e.g: "failed to validate capacity reservation group id : %w"